chalice/deploy/planner.py

```diff
@@ -96,12 +103,16 @@ def _plan_deletion(self,
                 apicall = models.APICall(
                     method_name='delete_function',
                     params={'function_name': resource_values['lambda_arn']},)
+                messages[id(apicall)] = (
+                    "Deleting function: %s\n" % resource_values['lambda_arn'])
                 plan.append(apicall)
             elif resource_values['resource_type'] == 'iam_role':
                 apicall = models.APICall(
                     method_name='delete_role',
                     params={'name': resource_values['role_name']},
                 )
+                messages[id(apicall)] = (
+                    "Deleting IAM role: %s\n" % resource_values['role_name'])
```

Comment:
In the old deployer, I think we prompted on whether we wanted to delete the role. Do we want to keep doing that?

Reply:
I'm not sure.  I was wondering if it made sense to have a more general pattern to this.  For example:

* Add a `plan` command that pretty prints the plan of the exact API calls we want to make
* Update the `deploy` command to pretty print the plan and confirm if we should proceed or not.

The thing I don't like about the previous approach was that we cherry picked API calls we thought were worth flagging and I'd like something more general.  It also meant that you couldn't know before hand whether or not a deploy was going to prompt you for something (making it harder to script).

Thoughts?

Reply:
I like the general idea. It definitely was weird that we would cherry-pick the api calls. As to the examples, I think I like the `plan` command the most, but instead what if we added a `--dryrun` or `--dry-run` flag that will pretty print the plan out? The reason I suggest this is that we could add the flag to both `deploy` and `delete`. If we just had a `plan` command, we would have to add an argument to differentiate what chalice command we are planning for. As for pretty printing the `deploy` command and confirming, would that be opt-in? I really like the ability to just run `chalice deploy` and it all works without further interactions.

Reply:
Here's a sketch of what we could do:

* By default, deploy/delete no longer prompt.  This makes it easier for the people scripting so they have a guarantee that we will never prompt by default.
* Add an option to `deploy/delete` (something like `--confirm`).  This would pretty print the plan and require confirmation from the user before executing the plan.  This is opt-in.
* Add a `--dry-run` flag to `deploy` and `delete`.  I think this would be a post-merge to `develop` feature.  It would essentially pretty print the plan just like `confirm`, and then exit.  We might even be able to add some sort of `FakeExecutor` that knows how to stub out API calls so you can actually test out the execution plan (i.e this would catch bugs where the planner emits a bad plan).